run_tests.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+#
+# A library containing community-based extension for the python-telegram-bot library
+# Copyright (C) 2021
+# The ptbcontrib developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""Helper script to run the test suites for ptbcontrib"""
+import subprocess  # nosec
+import sys
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import List
+
+from pygit2 import Repository
+
+root_path = Path(__file__).parent.resolve()
+ptbcontrib_path = root_path / 'ptbcontrib'
+contrib_paths = [
+    x for x in ptbcontrib_path.iterdir() if x.is_dir() and '__pycache__' not in x.name
+]
+contrib_names = [contrib.name for contrib in contrib_paths]
+
+
+def get_changed_contrib_names() -> List[str]:
+    """Get all changed files as compared to remote/main"""
+    repo = Repository(root_path)
+    main_branch = repo.lookup_branch('main')
+    if main_branch is None:
+        raise RuntimeError("Can't find `main` branch to compare to.")
+
+    file_paths = (patch.delta.new_file.path for patch in repo.diff(main_branch))
+    changed_contribs = set()
+
+    for filepath in file_paths:
+        if '__pycache__' not in filepath:
+            path_parents = (root_path / Path(filepath)).parents
+            for contrib_path in contrib_paths:
+                if contrib_path in path_parents:
+                    changed_contribs.add(contrib_path.name)
+
+    return list(changed_contribs)
+
+
+def run_tests(changed: bool, names: List[str]) -> int:
+    """Run the required tests and install requirements for each one"""
+    if changed:
+        names = get_changed_contrib_names()
+    elif not names:
+        names = contrib_names
+
+    exit_code = 0
+    for name in names:
+        try:
+            subprocess.check_call(  # nosec
+                [
+                    sys.executable,
+                    "-m",
+                    "pip",
+                    "install",
+                    "-r",
+                    str(ptbcontrib_path / name / "requirements.txt"),
+                ]
+            )
+            subprocess.check_call(  # nosec
+                [
+                    'pytest',
+                    '-v',
+                    str(root_path / 'tests' / f'test_{name}.py'),
```

Comment:
Can't we just do
```python
f"{root_path}/tests/test_{name}.py"
```
Same for above.

Reply:
that wouldn't work on windows … `path / path_or_str` does more or less what `os.path.join` does

Reply:
Oh i see, fine then.